integration-test/core/src/main/java/io/quarkiverse/resteasy/problem/ValidationExceptionsResource.java

```diff
@@ -121,4 +136,43 @@ public static final class ProgrammaticTestAddress {
         public String city;
     }
 
+    public static final class CustomInputBean {
+      @NotNull
+      @Valid
+      public CustomName name;
+    }
+
+    @ValidCustomName
+    public static final class CustomName {
+      public String code;
+    }
+
+    @Constraint(validatedBy = CustomNameValidator.class)
+    @Target({ ElementType.TYPE })
+    @Retention(RetentionPolicy.RUNTIME)
+    public @interface ValidCustomName {
+      String message() default "must match regex";
+
+      Class<?>[] groups() default {};
+
+      Class<? extends Payload>[] payload() default {};
+    }
+
+    public static class CustomNameValidator implements ConstraintValidator<ValidCustomName, CustomName> {
+      private final Pattern pattern = Pattern.compile("^[A-Z]{4}0[0-9]{6}$");
+
+      @Override
+      public boolean isValid(CustomName customName, ConstraintValidatorContext constraintValidatorContext) {
+        String code = customName.code;
+        constraintValidatorContext.disableDefaultConstraintViolation();
+
+        if (pattern != null && !pattern.matcher(code).matches()) {
+          constraintValidatorContext.buildConstraintViolationWithTemplate("must match \"" + pattern.pattern() + "\"")
+              .addConstraintViolation();
+          return false;
+        }
+        return true;
```

Comment:
You missed part of my comment/suggestion:
`pattern` is final now, so it's never null - check is redundant. Also, please either inline `code` variable, or move it right above the place where it is used for the first time. Right now the sections that do different things are tangled - first you prepare tmp variable, then call some modifier method on the context, then you use this variable later.

This should do:
```suggestion
        constraintValidatorContext.disableDefaultConstraintViolation();

        String code = customName.code;
        if (!pattern.matcher(customName.code).matches()) {
          constraintValidatorContext.buildConstraintViolationWithTemplate("must match \"" + pattern.pattern() + "\"")
              .addConstraintViolation();
          return false;
        }
        return true;
```